pettingzoo/utils/save_observation.py

```diff
@@ -23,7 +27,12 @@ def _check_observation_saveable(env, agent):
 
 # save the observation of an agent. If agent not specified uses env selected agent. If all_agents
 # then all agents in environment observation recorded.
-def save_observation(env, agent=None, all_agents=False, save_dir=os.getcwd()):
+def save_observation(
+    env: AECEnv,
```

Comment:
Can't this also be ParallelEnv?

Reply:
I don't think so, because the function calls `env.observe` which is only present in `AECEnv`

Reply:
Can you test this out to be sure? I feel like observe might be implicitly defined or something (I guess no last call though). Not sure why this is the case to be honest.

Reply:
I checked the class ParallelEnv and it doesn't define .observe() nor inherits from another class. I can allow the argument to be a ParallelEnv and ignore the pyright error.

Reply:
In that case then I think it’s fine, maybe make a note that observe() isn’t implemented by parallel envs

Reply:
Added a comment before calling env,observe.
https://github.com/LetteraUnica/PettingZoo/blob/type-hints/pettingzoo/utils/save_observation.py#L50